### DISCUSSION TITLE
How do I use keyboard shortcuts to scroll the preview content on the right side?

### DISCUSSION THREAD

**Discussion by gengdz** [ibhagwan/fzf-lua#1281] [Q&A] [answered]
When the file content is very long, I want to scroll down with `C-d` and scroll up with `C-f` What should I do?

![image](https://github.com/ibhagwan/fzf-lua/assets/41912568/ae2ec0d9-7318-4f98-99f0-28fb603b6c20)

**Comment by ibhagwan** [comment, marked as answer]
Default keys are `shift-up|down`, use `F1` for help, how to modify can be found by searching the discussions:
https://github.com/ibhagwan/fzf-lua/discussions?discussions_q=scroll

**Reply by gengdz** [reply]
The above shortcuts conflict with the input history. I would like to map Ctrl + Down to next-history. How should I do that?    
    
 wiki:  https://github.com/ibhagwan/fzf-lua/wiki#how-do-i-setup-input-history-keybinds
 ```lua
  fzf = {
          ["ctrl-d"] = "next-history",
          ["ctrl-u"] = "prev-history",
          ["ctrl-n"] = "down",
          ["ctrl-p"] = "up",
        },
```

I've tried `["ctrl-down"]` and `["C-Down"]` but neither work.

**Reply by ibhagwan** [reply]
Ctrl-down/up aren’t valid with fzf, see `man fzf` for what you can use.

**Reply by ibhagwan** [reply]
Actually, since you are using the builtin previewer you can use ctrl-up/down, define these under `keymap.builtin` and make sure to use neovim style binds I believe `<C-down>` should work but I didn’t test it.

**Reply by gengdz** [reply]
I found that C-j / C-k are also very convenient, I can use these for now.

**Reply by ibhagwan** [reply]
Great!

**Comment by sohanemon** [comment]
can we scroll left/right when wrap is disabled?

**Reply by ibhagwan** [reply]
> can we scroll left/right when wrap is disabled?

You can use a key bind to focus the preview window (any neovim win change bind or an fzf-lua `focus-preview`)l once you’re in the preview window scroll as you would in  any neovim win `zL` `zH`.
